dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/ActionPlanController.cs

```diff
@@ -149,20 +149,20 @@ public ActionResult Index()
         {
             if (id.HasValue)
             {
-                var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.ActionPlanTasks.GetByIdAsync(id.ToString()), personId, recordId);
+                var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.ActionPlanTasks.GetByIdAsync(id.Value), personId, recordId);
                 return View(response);
             }
 
-            var task = new ActionPlanTaskInstanceV2();
+            var task = new ActionPlanTaskInstance();
 
             if (planId.HasValue)
             {
-                task.AssociatedPlanId = planId.Value.ToString();
+                task.AssociatedPlanId = planId.Value;
             }
 
             if (objectiveId.HasValue)
             {
-                task.AssociatedObjectiveIds = new Collection<string> { objectiveId.Value.ToString() };
+                task.AssociatedObjectiveIds = new Collection<Guid?> { objectiveId.Value };
```

Comment:
Why are the objective IDs nullable Guids?

Reply:
Pretty sure that's a function of the autogen. If it's not marked "required", it wants to make it nullable. In this case the collection is required, but no way (that I know of) to specify that the items INSIDE the collection are required as well. Something to look into on the service side.